packages/dev/core/src/Lights/Shadows/shadowGenerator.ts

```diff
@@ -1413,6 +1413,8 @@ export class ShadowGenerator implements IShadowGenerator {
     protected _isReadyCustomDefines(defines: any, subMesh: SubMesh, useInstances: boolean): void {}
 
     private _prepareShadowDefines(subMesh: SubMesh, useInstances: boolean, defines: string[], isTransparent: boolean): string[] {
+        defines.push("#define SM_LIGHTTYPE_" + this._light.getClassName().toUpperCase());

```

Comment:
It is needed in the ray marching node when generating the shadow map, because the ray position/direction is not the same if the projection is orthographic or perspective.

For a bit of context, here's the basic ray marcher used in the demo PG:
```glsl
void raymarch(vec3 worldPosIn, vec3 worldCameraPos, float time, mat4 world, out vec4 worldPos, out vec4 worldNormal) {
    #if defined(SM_LIGHTTYPE_DIRECTIONALLIGHT)
        vec3 ro = worldPosIn;
        vec3 rd = normalize(lightDataSM);
    #else
        vec3 ro = worldCameraPos;
        vec3 rd = normalize(worldPosIn - ro);
    #endif
    float dO = 0.;
    mat4 invWorld = inverse(world);
    for(int i=0;i<100;i++)
    {
        vec4 p = vec4(ro + rd * dO, 1.0);
        float ds = sdf((invWorld * p).xyz);
        dO += ds;
        if(dO > 100. || ds < 0.01) break;
    }
    if (dO > 100.) discard;
    worldPos = vec4(ro + rd * dO, 1.);
    worldNormal = vec4(calcNormal((invWorld * worldPos).xyz), 0.);
}
```